### PR TITLE
PE-4982: Plausible custom events

### PR DESCRIPTION
**Pull request by karlprieb**
--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/5imd7d7t59db8

### CODE REVIEW

File: lib/authentication/login/views/login_page.dart [hunk 1]
```diff
@@ -1134,7 +1137,8 @@ class OnBoardingViewState extends State<OnBoardingView> {
                     wallet: widget.wallet,
                   ),
                 );
-            PlausibleEventTracker.track(event: PlausibleEvent.tutorialSkipped);
+            PlausibleEventTracker.trackPageview(
+                event: PlausiblePageView.tutorialSkipped);
```
Comment (matibat) [inline lib/authentication/login/views/login_page.dart:1141]: Should be a custom event.

File: lib/authentication/login/views/login_page.dart [hunk 2]
```diff
@@ -1186,11 +1190,14 @@ class OnBoardingViewState extends State<OnBoardingView> {
...
+      PlausibleEventTracker.trackPageview(
+          event: PlausiblePageView.tutorialsPage1);
     } else if (_currentPage == 1) {
-      PlausibleEventTracker.track(event: PlausibleEvent.tutorialsPage2);
+      PlausibleEventTracker.trackPageview(
+          event: PlausiblePageView.tutorialsPage2);
     } else if (_currentPage == 2) {
-      PlausibleEventTracker.track(event: PlausibleEvent.tutorialsPage3);
+      PlausibleEventTracker.trackPageview(
+          event: PlausiblePageView.tutorialsPage3);
```
Comment (matibat) [inline lib/authentication/login/views/login_page.dart:1200]: Should be custom events.

File: lib/components/profile_card.dart [hunk 3]
```diff
@@ -373,7 +373,8 @@ class __LogoutButtonState extends State<_LogoutButton> {
           arDriveAuth.logout().then(
             (value) {
               profileCubit.logoutProfile();
-              PlausibleEventTracker.track(event: PlausibleEvent.logout);
+              PlausibleEventTracker.trackPageview(
+                  event: PlausiblePageView.logout);
```
Comment (matibat) [inline lib/components/profile_card.dart:377]: Should be a custom event.

File: lib/pages/drive_detail/drive_detail_page.dart [hunk 4]
```diff
@@ -79,11 +79,11 @@ class _DriveDetailPageState extends State<DriveDetailPage> {
...
     if (widget.anonymouslyShowDriveDetail) {
-      PlausibleEventTracker.track(
-          event: PlausibleEvent.fileExplorerNonLoggedInUser);
+      PlausibleEventTracker.trackPageview(
+          event: PlausiblePageView.fileExplorerNonLoggedInUser);
     } else {
-      PlausibleEventTracker.track(
-          event: PlausibleEvent.fileExplorerLoggedInUser);
+      PlausibleEventTracker.trackPageview(
+          event: PlausiblePageView.fileExplorerLoggedInUser);
```
Comment (matibat) [inline lib/pages/drive_detail/drive_detail_page.dart:86]: Could be consolidated into a single event with custom properties.

File: lib/pages/no_drives/no_drives_page.dart [hunk 5]
```diff
@@ -19,12 +19,12 @@ class NoDrivesPage extends StatelessWidget {
...
-      PlausibleEventTracker.track(
-        event: PlausibleEvent.fileExplorerNonLoggedInUser,
+      PlausibleEventTracker.trackPageview(
+        event: PlausiblePageView.fileExplorerNonLoggedInUser,
       );
     } else {
-      PlausibleEventTracker.track(
-        event: PlausibleEvent.fileExplorerNewUserEmpty,
+      PlausibleEventTracker.trackPageview(
+        event: PlausiblePageView.fileExplorerNewUserEmpty,
```
Comment (matibat) [inline lib/pages/no_drives/no_drives_page.dart:27]: Could be consolidated into a single event with custom properties.

File: lib/turbo/topup/views/topup_modal.dart
Comment (matibat) [inline lib/turbo/topup/views/topup_modal.dart:1]: Should be custom events.

File: lib/utils/plausible_event_tracker.dart [hunk 6]
```diff
@@ -1,43 +1,134 @@
...
+      data: plausibleData,
+      eventData: eventData,
+    );
+  }
+
+  static Future<Map<String, dynamic>> _defaultEventProps() async {
+    Map<String, String> props = {
+      'Platform': AppPlatform.getPlatform().name,
+      'App Version': (await PackageInfo.fromPlatform()).version,
+    };
```
Comment (matibat) [inline lib/utils/plausible_event_tracker.dart:55]: We can benefit from an explicit type.

Comment (matibat) [reply lib/utils/plausible_event_tracker.dart:55]: Will every event track these default properties? An alternative is to track these only in the very first event we emit.

**Review by matibat — COMMENTED** [review]
Looking good so far. Left some comments.

**Review by thiagocarvalhodev — APPROVED** [review]

**Review by thiagocarvalhodev — APPROVED** [review]

**Pull request merged**